src/kixi/heimdall/handler.clj

```diff
@@ -97,14 +104,35 @@
       {:status 201 :body res}
       {:status 401 :body res})))
 
+(defn invalidate-refresh-token [session auth-conf refresh-token]
+  (if-let [unsigned (unsign-token auth-conf refresh-token)]
+    (let [user-uuid (java.util.UUID/fromString (:user-id unsigned))
+          refresh-token-data (refresh-token/find-by-user-and-issued session
+                                                                    user-uuid
+                                                                    (:iat unsigned))]
+      (do
+        (refresh-token/invalidate! session (:id refresh-token-data))
+        [true {:message "Invalidated successfully"}]))
+    [false {:message "Invalid or expired refresh token provided"}]))
+
+(defn invalidate-refresh-token-route [req]
+  (let [refresh-token (-> req :params :refresh-token)
+        [ok? res] (invalidate-refresh-token (:cassandra-session (:components req))
+                                            (:auth-conf req)
+                                            refresh-token)]
+    (if ok?
+      {:status 201 :body res}
```

Comment:
202 means accepted but not completed, so that's not right either.  Maybe 200 is the way to go. correcting ...